source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -73,6 +74,25 @@ std::vector<Http::HeaderUtility::HeaderData> headerMatchers(const T& matcher_pro
   return matchers;
 }
 
+// Transforms the proto list of 'auth_scopes' into a vector of std::string, also
+// handling the default value logic.
+std::vector<std::string>
+authScopesList(const Protobuf::RepeatedPtrField<std::string>& auth_scopes_protos) {
+  std::vector<std::string> scopes;
+
+  // If 'auth_scopes' is empty it must return a list with the default value.
+  if (auth_scopes_protos.empty()) {
+    scopes.emplace_back(DEFAULT_AUTH_SCOPE);
+  } else {
+    scopes.reserve(auth_scopes_protos.size());
+
+    for (const auto& scope : auth_scopes_protos) {
+      scopes.emplace_back(scope);
+    }
+  }
+  return scopes;
+}
+
```

Comment:
I would put this in the anonymous namespace, good practice to avoid symbol collision

Reply:
@snowp sorry but I might be missing something. This function is in the unnamed `namespace { `declared at line 35. Should be enough or I need to move it somewhere else?

Reply:
My bad then, I must have forgotten to scroll up!